crates/astria-conductor/src/celestia/latest_height_stream.rs

```diff
@@ -0,0 +1,63 @@
+use std::{
+    pin::Pin,
+    time::Duration,
+};
+
+use astria_eyre::eyre::{
+    Result,
+    WrapErr as _,
+};
+use celestia_client::{
+    celestia_rpc::HeaderClient as _,
+    jsonrpsee::http_client::HttpClient,
+};
+use futures::{
+    Future,
+    FutureExt as _,
+    Stream,
+    StreamExt as _,
+};
+use tokio_stream::wrappers::IntervalStream;
+
+pub(super) fn stream_latest_heights(
+    client: HttpClient,
+    poll_period: Duration,
+) -> LatestHeightStream {
+    let f = Box::new(move |_: tokio::time::Instant| {
+        let client = client.clone();
+        async move {
+            client
+                .header_network_head()
+                .await
+                .wrap_err("failed to fetch network head")
+                .map(|header| header.height().value())
+        }
+        .boxed()
+    });
+    let mut interval = tokio::time::interval(poll_period);
+    interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
```

Comment:
skipping is ok b/c conductor will fetch all blocks if a few heights are missed in the stream right?

Reply:
Yeah, so say the last block fetched is 10 and the latest celestia height we observed is 10 - in this case no new blocks will be fetched.

If the stream fails for 9 consecutive heights but then 10th one succeeds, then the latest observed height jumps from 10 to 20. In that moment the stream schedules 11 to 20 one after the other.